pandas/__init__.py

```diff
@@ -202,8 +202,40 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
```

Comment:
IMHO, the code in `pandas/compat/pyarrow.py` should set a value of `pa_not_found` (or something like that) and that should be used in the test.  When I read this code, I didn't understand why you were testing a version number of pyarrow, so I think adding an additional value would make it clear that we are testing for the presence of pyarrow here.

Reply:
Thx, simplified.